rust/src/treefile.rs

```diff
@@ -681,15 +681,21 @@ impl Treefile {
         s.to_string()
     }
 
-    /// Error out if the default is sqlite, but something else is configured.
+    /// Error out if the backend is ndb, which Fedora doesn't support.
     /// xref https://bugzilla.redhat.com/show_bug.cgi?id=1938928
     pub(crate) fn validate_rpmdb(&self) -> CxxResult<()> {
-        if let Some(rpmdb) = self.parsed.rpmdb.as_ref() {
-            if DEFAULT_RPMDB_BACKEND == RpmdbBackend::Sqlite && *rpmdb != DEFAULT_RPMDB_BACKEND {
-                return Err(anyhow!("Cannot write rpmdb backend: {:?}", rpmdb).into());
-            }
+        match self.parsed.rpmdb.as_ref() {
+            Some(&RpmdbBackend::Sqlite) | Some(&RpmdbBackend::Bdb) => Ok(()),
+            other => Err(anyhow!("Unsupported rpmdb backend: {:?}", other).into()),
         }
-        Ok(())
+    }
+
+    /// Returns true if the database backend is the same as the rpm-ostree default.
+    pub(crate) fn rpmdb_backend_is_default(&self) -> bool {
```

Comment:
I guess technically here we should instead check what we wrote vs what the target rpm defaults to. Not sure if there's an easy way to do that. Hmm, I think we could just look at the default value of `%_db_backend` in the rootfs?

I mean, hopefully we don't have another format transition like this in a while, so as long as the treefile directives match the target (and they basically have to at this point the way it happened), it's equivalent.

Reply:
I have a bigger patch cooking along the lines of https://github.com/coreos/rpm-ostree/pull/2990#issuecomment-879164167